lxd/storage_volumes_snapshot.go

```diff
@@ -625,6 +629,35 @@ func autoCreateCustomVolumeSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
 			return
 		}
 
+		// Get list of cluster nodes
+		nodes, err := cluster.List(d.State(), d.gateway)
+		if err != nil {
+			return
+		}
+
+		var availableNodeIDs []int64
+
+		for _, node := range nodes {
+			var nodeID int64
+
+			if node.Status == "Online" {
+				err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
+					info, err := tx.GetNodeByName(node.ServerName)
+					if err != nil {
+						return err
+					}
+
+					nodeID = info.ID
+					return nil
+				})
+				if err != nil {
+					continue
```

Comment:
@monstermunchkin not sure about continuing here, if there is an issue with retrieving info from the database we probably want to stop, rather than potentially risk going into non-clustered mode below and snapshotting on every node.

Reply:
Fixed